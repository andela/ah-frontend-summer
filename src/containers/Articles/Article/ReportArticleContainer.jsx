import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { reportArticle } from '../../../store/actions/async/ReportArticle';
import ReportArticle from '../../../components/UI/Article/ReportArticle';

export class ReportArticleContainer extends Component {
    state = {
        reason: ''
    };

    onChangeHandler = (e, { value }) => this.setState({ reason: value });

    onSubmitHandler = e => {
        e.preventDefault();
        const { 
            reportArticle, 
            match: { params: { slug } = {} } = {}, 
            history: { push } = {}
        } = this.props;
        const { reason } = this.state;
        const token = localStorage.getItem('token');
        const data = { reason }; 
        reportArticle({ token, data, slug, push });
    };

    render() {
        const { reason } = this.state;
        const { match: { params: { slug } = {} } = {} } = this.props;
        const reportArticleProps = {
            onSubmitHandler: this.onSubmitHandler,
            onChangeHandler: this.onChangeHandler,
            slug,
            reason
        };
        return <ReportArticle {...reportArticleProps}/>;
    }
}

const mapDispatchToProps = dispatch => {
    return {
        reportArticle: payload => dispatch(reportArticle(payload))
    };
};

ReportArticleContainer.propTypes = {
    reportArticle: PropTypes.func.isRequired,
    match: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired
};

export default connect(null, mapDispatchToProps)(ReportArticleContainer);
