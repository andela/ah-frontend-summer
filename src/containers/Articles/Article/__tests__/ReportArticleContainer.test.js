import React from 'react';
import { shallow } from 'enzyme';
import { ReportArticleContainer } from '../ReportArticleContainer';

describe('Report Article Container', () => {
    const reportArticleMock = jest.fn();
    const props = {
        reportArticle: reportArticleMock,
        match: { params: { slug: "fsd"} },
        history: {}
    }
    const wrapper = shallow(<ReportArticleContainer {...props} />);
    const wrapperInstance = wrapper.instance();

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('onChangeHandler updates value of reason in state', () => {
        wrapperInstance.onChangeHandler({}, { value: 'okay' });
        expect(wrapper.state().reason).toBe('okay');
    });

    it('onSubmithandler calls reportArticle function', () => {
        wrapperInstance.onSubmitHandler({ preventDefault: jest.fn() });
        expect(reportArticleMock).toHaveBeenCalled();
    });
});
