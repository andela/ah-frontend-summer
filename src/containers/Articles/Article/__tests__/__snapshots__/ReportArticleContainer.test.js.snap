// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report Article Container renders correctly 1`] = `
<Connect(ReportArticle)
  onChangeHandler={[Function]}
  onSubmitHandler={[Function]}
  reason=""
  slug="fsd"
/>
`;
